### ISSUE TITLE
TopNRowNumber fails when updating the memory usage

### ISSUE THREAD

**Issue by yingsu00** [prestodb/presto#9780]
```
presto:tiny> SELECT custkey, orderkey, rank FROM (
    SELECT custkey, orderkey, row_number() OVER (PARTITION BY custkey ORDER BY orderkey) AS rank FROM orders)
WHERE rank = 3;

Query 20180120_011051_00000_rajdf, FAILED, 4 nodes
http://localhost:8080/query.html?20180120_011051_00000_rajdf
Splits: 68 total, 47 done (69.12%)
CPU Time: 3.0s total,    5K rows/s,     0B/s, 9% active
Per Node: 0.2 parallelism, 1.09K rows/s,     0B/s
Parallelism: 0.9
0:03 [15K rows, 0B] [4.37K rows/s, 0B/s]

Query 20180120_011051_00000_rajdf failed: bytes cannot be negative
java.lang.IllegalArgumentException: bytes cannot be negative
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
	at com.facebook.presto.memory.context.SimpleLocalMemoryContext.setBytes(SimpleLocalMemoryContext.java:55)
	at com.facebook.presto.operator.OperatorContext$DecoratedLocalMemoryContext.setBytes(OperatorContext.java:632)
	at com.facebook.presto.operator.TopNRowNumberOperator.getPage(TopNRowNumberOperator.java:341)
	at com.facebook.presto.operator.TopNRowNumberOperator.getOutput(TopNRowNumberOperator.java:252)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:379)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:278)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:645)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:272)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:975)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.LegacyPrioritizedSplitRunner.process(LegacyPrioritizedSplitRunner.java:23)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:492)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by findepi** [comment]
@yingsu00 were you able to fix this?

**Comment by highker** [comment]
The fix will be superseded by #9686

**Comment by findepi** [comment]
@nezihyigitbasi's https://github.com/prestodb/presto/pull/9826 is merged
